src/one_dragon_qt/widgets/zoomable_image_label.py

```
@@ -1,55 +1,298 @@
+        image_x = (mouse_pos.x() - self.image_offset.x()) / self.scale_factor
+        image_y = (mouse_pos.y() - self.image_offset.y()) / self.scale_factor
+
+        # 更新缩放比例
+        if event.angleDelta().y() > 0:
+            self.scale_factor *= 1.1  # 放大
         else:
-            # 如果没有按住Ctrl，事件将交给父控件处理（例如，滚动条）
-            super().wheelEvent(event)
+            self.scale_factor /= 1.1  # 缩小
+
+        # 重新计算偏移量，使鼠标保持指向图像中的同一点
```

Ruff (0.12.2)

Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)